File: framework/languages/java/samples/java/DeliveryDispatch/Server/CustomerGateway/src/main/java/systems/zlink/samples/deliverydispatch/server/customergateway/CustomerActor.java
package systems.zlink.samples.deliverydispatch.server.customergateway;

import java.util.concurrent.CompletionStage;
import systems.zlink.framework.actors.ZLinkActor;
import systems.zlink.framework.actors.ZLinkActorContext;

public final class CustomerActor implements ZLinkActor {
    private final String actorId;
    private final ZLinkActorContext context;

    public CustomerActor(String actorId, ZLinkActorContext context) {
        this.actorId = actorId;
        this.context = context;
    }

    public String actorId() {
        return actorId;
    }

    @Override
    public ZLinkActorContext context() {
        return context;
    }

    public CompletionStage<Void> push(Object message) {
        return context.boundSession()
            .send(message)
            .submit();
    }
}

File: framework/languages/java/samples/kotlin/DeliveryDispatch/Server/CustomerGateway/src/main/kotlin/systems/zlink/samples/kotlin/deliverydispatch/server/customergateway/CustomerActor.kt
package systems.zlink.samples.kotlin.deliverydispatch.server.customergateway

import systems.zlink.framework.actors.ZLinkActor
import systems.zlink.framework.actors.ZLinkActorContext
import systems.zlink.framework.kotlin.kotlin

class CustomerActor(private val id: String, private val actorContext: ZLinkActorContext) :
    ZLinkActor {
    fun actorId(): String = id

    override fun context(): ZLinkActorContext = actorContext

    // --8<-- [start:doc-dd-bound-session-push]
    suspend fun push(message: Any) {
        try {
            actorContext.boundSession().kotlin().send(message).await()
        } catch (_: RuntimeException) {
            // Delivery notifications are best effort when the customer session is stale.
        }
    }
    // --8<-- [end:doc-dd-bound-session-push]
}
